drivers/media/i2c/imx477.c

```diff
@@ -1747,22 +1863,29 @@ static int imx477_start_streaming(struct imx477 *imx477)
 	/* Apply customized values from user */
 	ret =  __v4l2_ctrl_handler_setup(imx477->sd.ctrl_handler);
 	if (ret)
-		return ret;
+		goto error_pm_put;
 
 	/* Set vsync trigger mode: 0=standalone, 1=source, 2=sink */
-	tm = (imx477->trigger_mode_of >= 0) ? imx477->trigger_mode_of : trigger_mode;
```

Comment:
@naushir Naush I haven't changed this, but this hunk seems to revert 074f41ba54c77a607898ee9cba6994215912f640